src/System.Linq.Expressions/src/System/Runtime/CompilerServices/RuntimeOps.MergedRuntimeVariables.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.CompilerServices
+{
+    public partial class RuntimeOps
```

Comment:
Is it ok for build targets that only have the interpreter to start exposing this `RuntimeOps` type? Due to the sharing of code using a partial class, these build targets will see the type as being empty without any of the Quote`, `CreateRuntimeVariables`, `MergeRuntimeVariables` methods showing up. I could have included and called the latter from the interpreter, but it has the `[Obsolete]` attribute on it (likely to prevent users from calling it, given it's an infrastructure method).

There are three options here:

* Leave it as-is; this type doesn't have to show up in public facades
* Conditionally make it `internal` or `public` using an `#if`
* Move the inner classes to another place, doing away with the `RuntimeOps` parent type

The last option has a breaking change potential because instances of these types leak out of `RuntimeVariables` expressions upon evaluation, so users may have written code that relies on the result of calling `GetType`. Note that by removing the clones of `RuntimeVariables` and `MergedRuntimeVariables` in favor of the inner types in `RuntimeOps`, the runtime type of the `IRuntimeVariables` instance on both the compiler and interpreter will be the same now.